src/blocks/store/s3.clj

```diff
@@ -221,9 +223,12 @@
         (object->block client bucket prefix stats)
         ; Otherwise, upload block to S3.
         (let [object-key (id->key prefix (:id block))
-              ; TODO: support function which allows you to mutate the ObjectMetadata before it is sent in the putObject call?
               metadata (doto (ObjectMetadata.)
                          (.setContentLength (:size block)))]
+          (when set-sse-flag?
+            (.setSSEAlgorithm metadata ObjectMetadata/AES_256_SERVER_SIDE_ENCRYPTION))
+          (when mutate-put-metadata-fn
+            (mutate-put-metadata-fn metadata))
```

Comment:
It might be more useful to call this function with both the block store record and the current metadata. That way you could write a more general-purpose mutation function here that picked up settings from the store level. Basically, data is easier to reason about (and work with) than function closures.